src/Polly.Core/Simmy/MonkeyStrategy.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Polly.Simmy.Utils;
+
+namespace Polly.Simmy;
+
+#pragma warning disable CA1031 // Do not catch general exception types
+#pragma warning disable S3928 // Custom ArgumentNullException message
+
+/// <summary>
+/// Contains common functionality for chaos strategies which intentionally disrupt executions - which monkey around with calls.
+/// </summary>
+public abstract class MonkeyStrategy : ResilienceStrategy
+{
+    private readonly Func<double> _randomizer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MonkeyStrategy"/> class.
+    /// </summary>
+    /// <param name="options">The chaos strategy options.</param>
+    protected MonkeyStrategy(MonkeyStrategyOptions options)
+    {
+        Guard.NotNull(options);
+        Guard.NotNull(options.Randomizer);
+
+        if (options.InjectionRate is null && options.InjectionRateGenerator is null)
```

Comment:
You can drop this check (and one bellow) if you make `InjectionRate` non-nullable with some default and `InjectionRateGenerator` nullable.